marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/security/getAuthorities.sjs

```diff
@@ -15,14 +15,49 @@
  */
 'use strict';
 
-// TODO This is just stubbed out, a real implementation is needed.
-
-let response = {
-  "authorities": [
-    "canInstallDataHub",
-    "canReadLoadData",
-    "canWriteLoadData"
-  ]
+const Artifacts = require('/data-hub/5/artifacts/core.sjs');
+const roleIdToName = (roleID) => xdmp.roleName(roleID);
+const currentRoles = xdmp.getCurrentRoles().toArray().map(String);
+let currentRoleNames = currentRoles.map(roleIdToName);
+
+const defaultManageRoles = ['admin', 'data-hub-environment-manager'].map((roleName) => String(xdmp.role(roleName)));
```

Comment:
This is a really good point - this needs a quick test, but if a user has role "apple", and "apple" inherits "manage-admin", then when you get the list of the current user's roles, you won't get "manage-admin" back, you'll only get "admin". 

The story needs to account for this scenario - e.g. a user that inherits manage-admin, but not directly. It may be better to check to see if the user has the "manage-admin" privilege, I just don't know off the top of my head if there's an easy way to get "all" roles and privileges, including inherited ones.

Reply:
I take that back - xdmp.getCurrentRoles() (and xdmp.roleRoles()) do return all inherited roles. 

@hao1st You don't need to check for "data-hub-environment-manager" then.  If a user has that role, then xdmp.getCurrentRoles() will return both "manage-admin" and "security".

The only role that requires special handling is "admin", since for a user with only "admin", xdmp.getCurrentRoles() will only return "admin".

Reply:
@rjrudin if the user has only "data-hub-portal-security-admin" role,  should we give "canInstallDataHub" authority?

Reply:
@ryanjdew If the user has "admin" role, all the canRead{artifactType} and canWrite{artifactType} are false. Only "canInstallDataHub" is the authortity. It does not seem right. I mean that "cores.sjs" also needs to do special handling for "admin" role?

Reply:
We don't care if the user has "data-hub-portal-security-admin" - that role is one of an infinite number of ways that a user may inherit manage-admin and security. All we care about is if the user has both manage-admin and security.

Reply:
Re: the admin role - IMO, the design should be that if the user has the admin role, then the user gains all authorities. That seems like the expected and intuitive approach, along with being consistent with how the admin role works within ML.

Reply:
@rjrudin the following response for "admin" is correct? As we talked before, we will remove the roles once UI starts to respect authorities. 
```
{
"authorities": [
"canInstallDataHub"
], 
"roles": [
"data-hub-environment-manager", 
"data-hub-mapping-reader", 
"data-hub-module-writer", 
"data-hub-admin-role", 
"data-hub-module-reader", 
"data-hub-entity-model-reader", 
"data-hub-explorer-architect", 
"data-hub-security-admin", 
"data-hub-flow-reader", 
"data-hub-step-definition-writer", 
"data-hub-operator", 
"data-hub-mapping-writer", 
"data-hub-admin", 
"data-hub-load-data-writer", 
"data-hub-job-reader", 
"data-hub-job-internal", 
"data-hub-monitor", 
"data-hub-entity-model-writer", 
"data-hub-developer", 
"data-hub-portal-security-admin", 
"data-hub-load-data-reader", 
"data-hub-step-definition-reader", 
"data-hub-flow-writer"
]
}
```

Reply:
If that's the complete set of authorities and DHF-related roles, then yes, it's correct. With the assumption that roles goes away very soon.